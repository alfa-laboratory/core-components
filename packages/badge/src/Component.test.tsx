import React from 'react';
import { render } from '@testing-library/react';

import { Badge } from './index';

describe('Badge', () => {
    describe('Snapshots tests', () => {
        it('should match snapshot', () => {
            const { container } = render(<Badge view='count' content={100} />);

            expect(container).toMatchSnapshot();
        });
    });

    it('should forward ref', () => {
        const ref = jest.fn();
        const dataTestId = 'test-id';
        const { getByTestId } = render(<Badge view='count' ref={ref} dataTestId={dataTestId} />);

        expect(ref.mock.calls).toEqual([[getByTestId(dataTestId)]]);
    });

    it('should set `data-test-id` attribute', () => {
        const dataTestId = 'test-id';
        const { getByTestId } = render(<Badge view='count' dataTestId={dataTestId} />);

        expect(getByTestId(dataTestId).tagName).toBe('DIV');
    });

    describe('Classes tests', () => {
        it('should set `size` class', () => {
            const size = 's';
            const { container } = render(<Badge view='count' size={size} />);

            expect(container.firstElementChild).toHaveClass(size);
        });

        it('should set `view` class', () => {
            const view = 'count';
            const { container } = render(<Badge view={view} />);

            expect(container.firstElementChild).toHaveClass(view);
        });

        it('should set `positive` class if `iconColor` prop is `positive`', () => {
            const iconColor = 'positive';
            const { container } = render(<Badge view='icon' iconColor={iconColor} />);

            expect(container.firstElementChild).toHaveClass(iconColor);
        });

        it('should set `isHidden` class if `content` prop lower than 0', () => {
            const { container } = render(<Badge view='count' content={0} />);

            expect(container.firstElementChild).toHaveClass('isHidden');
        });

        it('should set `dot` class without `content` prop', () => {
            const { container } = render(<Badge view='count' />);

            expect(container.firstElementChild).toHaveClass('dot');
        });
    });

    it('should contain `99+` if content is bigger than 99', () => {
        const { container } = render(<Badge view='count' content={100} />);

        expect(container).toHaveTextContent('99+');
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<Badge view='count' />);

        expect(unmount).not.toThrowError();
    });
});
