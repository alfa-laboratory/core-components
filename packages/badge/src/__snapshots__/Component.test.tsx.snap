// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Snapshots tests should match snapshot 1`] = `
<div>
  <div
    class="component m count"
  >
    99+
  </div>
</div>
`;
